packages/flutter_tools/lib/src/commands/build_apk.dart

```diff
@@ -100,6 +100,8 @@ class BuildApkCommand extends BuildSubCommand {
       printStatus('Learn more on:  https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split',indent: 8);
     }
     await androidBuilder.buildApk(
+      // A more intuitive behavior could be something along the lines of:
+      // FlutterProject.fromPath(findProjectRoot(targetFile))
```

Comment:
Are you suggesting that we should change `FlutterProject.current()` for a more verbose version?

Reply:
Yes -- there's a difference between `current()` and `targetFile`. E.g. when I have Flutter projects A and B and run

```
A$ flutter build ../B/lib/main.dart
```

from within directory of the project A, the current behavior is to build `../B/lib/main.dart` in the context of project A.

I have moved this proposal into a separate PR.